tests/tests_pytorch/loggers/test_tensorboard.py

```diff
@@ -275,7 +275,17 @@ def training_step(self, *args):
 def test_tensorboard_finalize(summary_writer, tmpdir):
     """Test that the SummaryWriter closes in finalize."""
     logger = TensorBoardLogger(save_dir=tmpdir)
+    assert logger._experiment is None
```

Comment:
Is there a use-case where user will also want to check if the `experiment` is None? Because in that case we wouldn't want them to use these protected objects/variables. If it's only for tests, then it's totally fine.

Reply:
@krshrimali since this is a test, it fine to assert the internals here. Not all statements in a test does necessarily reflect how a user would interact with the object. 

The user would access logger.experiment or other public attributes.

Reply:
@awaelchli Sorry if I wasn't clear with my question, I understand it's okay to assert the internals in test case.

My question is - will a user ever want to check if the `experiment` is None? If yes, anyway they can do it without checking the internals? More of a question then a blocker review to the PR.

Reply:
I don't think they want to, and they shouldn't. They expect experiment not to be None. Which is why when the user calls logger.experiment, we create one if one does not exist (we create it lazily).

Reply:
Got it, thanks @awaelchli!